icicle-core/src/main/resources/id-generation.lua

```diff
@@ -5,16 +5,21 @@ local logical_shard_id_key = 'icicle-generator-logical-shard-id'
 local max_sequence = tonumber(KEYS[1])
 local min_logical_shard_id = tonumber(KEYS[2])
 local max_logical_shard_id = tonumber(KEYS[3])
+local num_ids = tonumber(KEYS[4])
 
 if redis.call('EXISTS', lock_key) == 1 then
   redis.log(redis.LOG_INFO, 'Icicle: Cannot generate ID, waiting for lock to expire.')
   return redis.error_reply('Icicle: Cannot generate ID, waiting for lock to expire.')
 end
 
-local sequence = redis.call('INCR', sequence_key)
+--[[
+Increment by a set number, this can
+--]]
+local end_sequence = redis.call('INCRBY', sequence_key, num_ids)
+local start_sequence = end_sequence - num_ids
```

Comment:
Do the `+1` here (otherwise this isn't the start of the sequence as the name suggests).